src/network/message.rs

```diff
@@ -37,21 +37,51 @@ use consensus::encode::MAX_VEC_SIZE;
 #[derive(PartialEq, Eq, Clone, Debug)]
 pub struct CommandString(Cow<'static, str>);
 
+impl CommandString {
+    /// Convert from various string types into a [CommandString].
+    ///
+    /// Supported types are:
+    /// - `&'static str`
+    /// - `String`
+    /// - `&str`
+    ///
+    /// Returns an empty error if and only if the string is
+    /// larger than 12 characters in length.
+    pub fn from_str<S: Into<Cow<'static, str>>>(s: S) -> Result<CommandString, ()> {
```

Comment:
Some methods rely on `FromStr` implementation to do a conversion; for instance if use `clap` for parsing commands from command-line parameters (which may be a case for cli tools operating with bitcoin P2P network). So I propose also to add `impl FromStr for CommandString` and from it call this method directly